numpy/array_api/linalg.py

```diff
@@ -0,0 +1,346 @@
+from __future__ import annotations
+
+from ._dtypes import _floating_dtypes, _numeric_dtypes
+from ._array_object import Array
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ._typing import Literal, Optional, Sequence, Tuple, Union
+
+from typing import NamedTuple
+
+import numpy.linalg
+import numpy as np
+
+class eighresult(NamedTuple):
+    eigenvalues: Array
+    eigenvectors: Array
+
+class qrresult(NamedTuple):
+    q: Array
+    r: Array
+
+class slogdetresult(NamedTuple):
+    sign: Array
+    logabsdet: Array
+
+class svdresult(NamedTuple):
+    u: Array
+    s: Array
+    vh: Array
```

Comment:
Is there any reason for not going with camel-case names here?

Reply:
The field names are part of the spec. If you think those should be changed you should bring it up in the spec repo. The class name was chosen by me. We can change it to something else if you feel that would be better.

Reply:
Ah yes, to clarify: my comment was aimed at the class names, not the fields names. 
I was aiming at something like `SVDResult` instead of `svdresult` for example.

Reply:
Yes, we can change that. I've seen lowercase convention used sometimes for namedtuples, but I think there's not a consistent pattern. Even in NumPy itself it seems like both conventions are used in some places.